tests/components/wallbox/test_init.py

```diff
@@ -0,0 +1,165 @@
+    '{"charging_power": "XX","max_available_power": "xx","charging_speed": 0,"added_range": "xx","added_energy": "XX"}'
+)
+
+
+async def test_wallbox_setup_entry(hass: HomeAssistantType):
+    """Test Wallbox Setup."""
+    with requests_mock.Mocker() as m:
+        m.get(
+            "https://api.wall-box.com/auth/token/user",
+            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":200}',
+            status_code=200,
+        )
+        m.get(
+            "https://api.wall-box.com/chargers/status/12345",
+            text='{"Temperature": 100, "Location": "Toronto", "Datetime": "2020-07-23", "Units": "Celsius"}',
+            status_code=200,
+        )
+        assert await wallbox.async_setup_entry(hass, entry)
+
+    with requests_mock.Mocker() as m, raises(ConnectionError):
+        m.get(
+            "https://api.wall-box.com/auth/token/user",
+            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":404}',
+            status_code=404,
+        )
+        assert await wallbox.async_setup_entry(hass, entry) is False
+
+
+async def test_wallbox_unload_entry(hass: HomeAssistantType):
+    """Test Wallbox Unload."""
+    hass.data[DOMAIN] = {"connections": {entry.entry_id: entry}}
+
+    assert await wallbox.async_unload_entry(hass, entry)
+
+    hass.data[DOMAIN] = {"fail_entry": entry}
+
+    with pytest.raises(KeyError):
+        await wallbox.async_unload_entry(hass, entry)
+
+
+async def test_get_data(hass: HomeAssistantType):
+    """Test hub class, get_data."""
+
+    station = ("12345",)
+    username = ("test-username",)
+    password = "test-password"
+
+    hub = wallbox.WallboxHub(station, username, password, hass)
+
+    with requests_mock.Mocker() as m:
+        m.get(
+            "https://api.wall-box.com/auth/token/user",
+            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":200}',
+            status_code=200,
+        )
+        m.get(
+            "https://api.wall-box.com/chargers/status/('12345',)",
+            json=test_response,
+            status_code=200,
+        )
+        assert await hub.async_get_data()
+
+
+async def test_get_data_rounding_error(hass: HomeAssistantType):
+    """Test hub class, get_data with rounding error."""
+
+    station = ("12345",)
+    username = ("test-username",)
+    password = "test-password"
+
+    hub = wallbox.WallboxHub(station, username, password, hass)
+
+    with requests_mock.Mocker() as m:
+        m.get(
+            "https://api.wall-box.com/auth/token/user",
+            text='{"jwt":"fakekeyhere","user_id":12345,"ttl":145656758,"error":false,"status":200}',
+            status_code=200,
+        )
+        m.get(
+            "https://api.wall-box.com/chargers/status/('12345',)",
+            json=test_response_rounding_error,
+            status_code=200,
+        )
+        assert await hub.async_get_data()
```

Comment:
Patch the `pypi` code and setup the integration using the standard interfaces instead of calling it directly

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations